### PR TITLE
Add more tests for lexer.

### PR DESCRIPTION
**Pull request by fernape**
* Unterminated string.
 * Unknown token.

### CODE REVIEW

File: internal/parse/lexer_test.go [hunk 1]
```diff
@@ -83,6 +83,29 @@ FROM person`
...
 }
 
+func TestLexerUnterminatedString(t *testing.T) {
+	stmt := `select 's`
+
+	tokens := tokensForStatement(stmt)
+
+	assert.Equal(t, STRING, tokens[1].Type)
+	assert.Equal(t, "'s", tokens[1].Literal)
+	assert.Equal(t, 8, tokens[1].Pos.Column)
```
Comment (benhoyt) [inline internal/parse/lexer_test.go:93]: Should we not test for an error at the end as well? Presumably unterminated string is an error?

File: internal/parse/lexer_test.go [hunk 2]
```diff
@@ -83,6 +83,29 @@ FROM person`
 	}
 }
 
+func TestLexerUnterminatedString(t *testing.T) {
+	stmt := `select 's`
+
+	tokens := tokensForStatement(stmt)
+
+	assert.Equal(t, STRING, tokens[1].Type)
```
Comment (benhoyt) [inline internal/parse/lexer_test.go:91]: I'm curious why the arguments are reversed (here and in other places)? Normally you'd say `assert.Equal(t, tokens[1].Type, STRING)` so it reads more like the English order: "assert that tokens[1]'s Type field equals STRING".

File: internal/parse/lexer_test.go (hunk 1 above)
Comment (fernape) [reply internal/parse/lexer_test.go:93]: The lexer does not return errors. [Everything](https://github.com/canonical/sqlair/blob/1a38304ffea770462f250f3e655132e3e58dd0b6/internal/parse/lexer.go#L129) is handled [downstream](https://github.com/canonical/sqlair/blob/1a38304ffea770462f250f3e655132e3e58dd0b6/internal/parse/lexer.go#L166)

File: internal/parse/lexer_test.go (hunk 2 above)
Comment (fernape) [reply internal/parse/lexer_test.go:91]: The second parameter is the expected value and the third one the actual value.
E.g:

```
assert.Equal(t, 1, 2)
```

```
--- FAIL: TestLexerUnterminatedString (0.00s)
    lexer_test.go:94: 
                Error Trace:    lexer_test.go:94
                Error:          Not equal: 
                                expected: 1
                                actual  : 2
                Test:           TestLexerUnterminatedString
```

Comment (benhoyt) [reply internal/parse/lexer_test.go:91]: Oh, interesting, I hadn't realized that -- thanks!

**Review by manadart — APPROVED** [review]
Nice; thanks.

**Pull request merged**